contracts/extensions/contracts/MilestonesExtension.sol

```diff
@@ -0,0 +1,189 @@
+    }
+
+    /// @notice Review a pending milestone submitted by the acceptedRecipientId.
+    /// @dev Emits a 'MilestoneStatusChanged()' event.
+    /// @param _milestoneStatus New status of the milestone
+    function reviewMilestone(Status _milestoneStatus) external virtual {
+        _validateReviewMilestone(msg.sender, _milestoneStatus);
+        // Check if the milestone status is pending
+
+        milestones[upcomingMilestone].status = _milestoneStatus;
+
+        emit MilestoneStatusChanged(upcomingMilestone, _milestoneStatus);
+
+        if (_milestoneStatus == Status.Accepted) {
+            upcomingMilestone++;
+        }
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    function _setProposalBid(address _bidderId, uint256 _proposalBid) internal virtual {
+        if (_proposalBid > maxBid) {
+            // If the proposal bid is greater than the max bid this will revert
+            revert MilestonesExtension_EXCEEDING_MAX_BID();
+        } else if (_proposalBid == 0) {
+            // If the proposal bid is 0, set it to the max bid
+            _proposalBid = maxBid;
+        }
+
+        bids[_bidderId] = _proposalBid;
+        emit SetBid(_bidderId, _proposalBid);
+    }
+
+    function _validateSetMilestones(address _sender) internal virtual {
+        _checkOnlyPoolManager(_sender);
+        if (milestones.length > 0) {
+            if (milestones[0].status != Status.None) revert MilestonesExtension_MILESTONES_ALREADY_SET();
+            delete milestones;
+        }
+    }
+
+    function _validateSubmitUpcomingMilestone(address _sender) internal virtual {
+        // Check if the 'msg.sender' is the 'acceptedRecipientId'
+        if (acceptedRecipientId != _sender) revert MilestonesExtension_INVALID_SUBMITTER();
+
+        // Check if a submission is ongoing to prevent front-running a milestone review.
+        if (milestones[upcomingMilestone].status == Status.Pending) revert MilestonesExtension_MILESTONE_PENDING();
+    }
+
+    function _validateReviewMilestone(address _sender, Status _milestoneStatus) internal virtual {
+        _checkOnlyPoolManager(_sender);
+        if (_milestoneStatus == Status.None) revert MilestonesExtension_INVALID_MILESTONE_STATUS();
+        if (milestones[upcomingMilestone].status != Status.Pending) revert MilestonesExtension_MILESTONE_NOT_PENDING();
+    }
+
+    /// @notice Increase max bid for RFP pool
+    /// @param _maxBid The new max bid to be set
+    function _increaseMaxBid(uint256 _maxBid) internal {
+        // make sure the new max bid is greater than the current max bid
+        if (_maxBid < maxBid) revert MilestonesExtension_AMOUNT_TOO_LOW();
+
+        maxBid = _maxBid;
+
+        // emit the new max mid
+        emit MaxBidIncreased(maxBid);
+    }
+
+    function _getMilestonePayout(uint256 _milestoneId) internal view virtual returns (uint256) {
```

Comment:
This internal is never used.

Reply:
I expect this to be used in `_distribute()`